modules/calib3d/test/test_solvepnp_ransac.cpp

```diff
@@ -383,6 +378,22 @@ class CV_solvePnPRansac_Test : public cvtest::BaseTest
         {
             for (int method = 0; method < SOLVEPNP_MAX_COUNT; method++)
             {
+                // SOLVEPNP_IPPE need planar object
+                if (!planar && method == SOLVEPNP_IPPE)
+                {
+                    cout << "mode: " << printMode(mode) << ", method: " << printMethod(method) << " -> "
+                         << "Skip for non-planar object" << endl;
+                    continue;
+                }
+
+                // SOLVEPNP_IPPE_SQUARE need planar tag object
+                if (!planarTag && method == SOLVEPNP_IPPE_SQUARE)
+                {
+                    cout << "mode: " << printMode(mode) << ", method: " << printMethod(method) << " -> "
+                         << "Skip for non-planar tag object" << endl;
+                    continue;
+                }
+
```

Comment:
It's even better to throw `SkipTestException`. It's handled by test system.

Reply:
Here one test is for many methods, so it seems that throwing exception will abort whole test and following methods will not be tested